include/irr/asset/IDescriptorSet.h

```diff
@@ -167,12 +169,15 @@ class IEmulatedDescriptorSet
 			m_descriptors = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<typename IDescriptorSet<LayoutType>::SDescriptorInfo> >(descriptorCount);
 			// set up all offsets
 			prevBinding = 0u;
+            uint32_t i = 0u;
 			for (auto it=m_bindingInfo->begin(); it!=m_bindingInfo->end(); it++)
 			{
-				if (it->offset < descriptorCount)
-					prevBinding = it->offset;
+                if (it->offset < descriptorCount) {
+                    prevBinding = it->offset;
+                    i = 0u;
+                }
 				else
-					it->offset = prevBinding;
+					it->offset = prevBinding + (++i);
```

Comment:
this is all fucked

Reply:
I should have called `prevBinding` `prevOffset` instead and it would be less misleading